src/sentry/static/sentry/app/components/dropdownButton.jsx

```diff
@@ -3,28 +3,35 @@ import React from 'react';
 import styled from 'react-emotion';
 import Button from 'app/components/button';
 import InlineSvg from 'app/components/inlineSvg';
+import {omit} from 'lodash';
 
-const DropdownButton = ({isOpen, children, ...props}) => {
-  return (
-    <StyledButton isOpen={isOpen} {...props}>
-      {children}
-      <StyledChevronDown />
-    </StyledButton>
-  );
-};
-
-DropdownButton.displayName = 'DropdownButton';
-DropdownButton.propTypes = {
-  isOpen: PropTypes.bool,
-};
+class DropdownButton extends React.Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+  };
+  render() {
+    const {isOpen, children, ...otherProps} = this.props;
+    return (
+      <StyledButton isOpen={isOpen} {...otherProps}>
+        {children}
+        <StyledChevronDown />
+      </StyledButton>
+    );
+  }
+}
 
 const StyledChevronDown = styled(props => (
   <InlineSvg src="icon-chevron-down" {...props} />
 ))`
   margin-left: 0.33em;
 `;
 
-const StyledButton = styled(({isOpen, ...props}) => <Button {...props} />)`
+const StyledButton = styled(
+  React.forwardRef((props, ref) => {
+    const forwardProps = omit(props, ['isOpen']);
```

Comment:
i think you can just to `{...props}` below instead of `{...forwardProps}` and it will pull `isOpen` from the object because of the destructuring, no?

Reply:
I can but then there is an eslint error about the unused variable that I would need to suppress.

Reply:
gosh darn it will that is annoying. for some reason I thought we used this pattern in other places but maybe not.